src/System.CommandLine.Tests/OptionTests.cs

```diff
@@ -346,5 +346,38 @@ public void Option_of_enum_can_limit_enum_members_as_valid_values()
                 .Should()
                 .BeEquivalentTo(new[] { $"Argument 'Fuschia' not recognized. Must be one of:\n\t'Red'\n\t'Green'" });
         }
+
+        [Fact]
+        public void Every_option_can_provide_a_handler_and_it_takes_precedence_over_command_handler()
```

Comment:
What if two different options having handlers are both present?

Reply:
We have two possibilities:

- stop on the first active option
- let the last one to be the effective (this is what we already do for commands)

@jonsequitur choosing the last one is the most intuitive way for me. What do you think?

Reply:
We could also make it a parse error. Of course certain options (like `-h`) ignore that and the action still runs.